### PR TITLE
feat: DIDComm Outbound Transport - Support for WebSocket

### LINKED ISSUE #827: DIDComm Outbound Transport - Support for WebSocket
Aries framework Go should support WebSocket as one of the DIDComm transport. - https://github.com/hyperledger/aries-rfcs/tree/master/features/0025-didcomm-transports#websocket.

### PR DESCRIPTION
**Pull request by rolsonquadras**
Closes #827 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/didcomm/transport/ws/outbound.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	return &OutboundClient{}
+}
+
+// Send sends a2a data via WS.
+func (cs *OutboundClient) Send(data []byte, url string) (string, error) {
+	if url == "" {
+		return "", errors.New("url is mandatory")
+	}
+
+	client, _, err := websocket.DefaultDialer.Dial(url, nil)
```
Comment (troyronda) [inline pkg/didcomm/transport/ws/outbound.go]: dial on every Send?

Comment (rolsonquadras) [reply pkg/didcomm/transport/ws/outbound.go]: https://github.com/hyperledger/aries-rfcs/tree/master/features/0025-didcomm-transports#websocket "Websockets are considered transmit only by default. Messages will only flow from the agent that opened the socket." -- this is what the RFC says.

For WASM requirement, we have to figure out a way to keep connections in a pool or something when they connect to the server (inbound) and can be used to send/push the messages without creating a socket (also, wouldn't be possible to dial as wasm will not have an endpoint).

Comment (troyronda) [reply pkg/didcomm/transport/ws/outbound.go]: Yes. Probably a TTL scheme. I guess a TODO :).

Comment (rolsonquadras) [reply pkg/didcomm/transport/ws/outbound.go]: I didn't add TODO as it's a separate feature and might need to revisit this based on the solution.

**Pull request merged**